src/dmd/dmodule.d

```diff
@@ -377,6 +377,7 @@ extern (C++) final class Module : Package
      * ownership of template instantiation.
      */
     Module importedFrom;
+    bool isCompiledImport;
```

Comment:
This PR determines if an import should be compiled earlier than before (right after parsing the module declaration instead of after parsing the whole file).  This way, we know if we are going to compile unittests before parsing the file.  To that end, I found out that the parser depends on `importedFrom` being set correctly meaning we can no longer "hijack" this field by resetting it when an imported module is compiled.  The solution is to add a new field that indicates an imported module should be compiled, and to leave the `importedFrom` field intact so it can be used during the parse stage.